app/ui/labels.py

```diff
@@ -174,14 +181,27 @@ def on_create_click(self, name, group, rank, condition, is_dynamic):
                 "labels",
                 "create",
             )
-            outcome = self.snowflake.call(
-                "ADMIN.CREATE_LABEL", name, group, rank, condition, is_dynamic
-            )
-
-            if outcome is None:
-                self.session.set_toast("New label created.")
-                self.session.do_list()
-                return
+            try:
+                with with_session(self.snowflake) as sf:
```

Comment:
I need to rename the function to avoid the double "with", but what are you thoughts about this approach to the `context={..}` we could have used with pydantic 2.3?

Reply:
i like it

Reply:
I want to try and add this as an annotation so we don't have to manually do the `with` but thats cosmetic only

Reply:
Great, thanks! I added a note for the decorator in #234 to pick up later.